src/components/Fetch/index.js

```diff
@@ -146,20 +152,14 @@ export class Fetch extends React.Component {
 
   updateURL(bookmarks, filters) {
     const parsedParams = this.getUrlParams();
-    const searchParams = new URLSearchParams();
+    const locationSearch = this.props.location.search;
+    const parsed = queryString.parse(locationSearch === '' ? this.props.location.hash : locationSearch);
 
-    // make url match this state
-    let nextUrl = '';
-    if (!this.urlInput || !this.urlInput.value) {
-      nextUrl = '/lobster/build/' + parsedParams.build + '/test/' + parsedParams.test;
-    } else {
-      searchParams.append('url', this.urlInput.value);
-    }
     for (let i = 0; i < filters.length; i++) {
-      searchParams.append('f', this.makeFilterURLString(filters[i]));
+      parsed.f = this.makeFilterURLString(filters[i]);
     }
     if (parsedParams.scrollLine) {
-      searchParams.append('scroll', parsedParams.scrollLine);
+      parsed.scroll = parsedParams.scrollLine;
     }
     if (bookmarks.length > 0) {
```

Comment:
Your original approach of bookmarks.length > 2 works and is probably the right call. Let's go with it!